xayn-ai-ffi-c/src/reranker/ai.rs

```diff
@@ -175,6 +176,55 @@ impl CXaynAi {
     }
 }
 
+/// Initializes the global thread pool. The thread pool is used by the AI to
+/// parallel some of its tasks.
+///
+/// The number of threads used by the pool depends on `num_cpus`:
+///
+/// On a single core system the thread pool consists of only one thread.
+/// On a multicore system the thread pool consists of (the number of logical cores - 1) threads.
+///
+/// # Error
+/// - If the initialization of the thread pool has failed.
+/// - An unexpected panic happened during the initialization of the thread pool.
+///
+/// # Safety
```

Comment:
I don't think you need to mention this nor does this function need to be `unsafe`.

By accepting a `Option<&mut CError>` we "enforce" that the safety aspects are upheld the only way to to brake this is to use "incorrect unsafe code" (here unsafe C code)